### ISSUE TITLE
Invalid IP error

### ISSUE THREAD

**Issue by Vincent-aCRM** [sugarcrm/IPRestrictionManager#3]
This is strange, same module installed on some Sugar Enterprise instances works fine, bit on few others we get this error, even though they are all the exact same version 9.x

```
**Invalid Credentials** You are attempting to authenticate from an invalid IP
(2a00:1098:0:80:1000:61:1:1) or platform 
(base).
```

* See screenshot
Any ideas to point me in right direction appreciated.
![aCloudDemo_IPrestriction-error](https://user-images.githubusercontent.com/56155559/70328245-75ad2600-1830-11ea-8abe-1bbc8d160352.jpg)

**Comment by geraldclark** [comment]
Hey Vincent,

That error is just what's thrown when the ip or platform don't match what's expected for that user.

Are you sure the ip restriction list is the same between the instances? Also, are there any notable differences between the stacks that work and the stacks that don't - like the php version?

**Comment by Vincent-aCRM** [comment]
Hi Gerald, thanks for reaching out.
Stack could be different.
Shall I provide a phpinfo.html for one of the non working instances if that helps?

**Comment by geraldclark** [comment]
if you can provide an export of your IP restrictions and just let me know the PHP versions for the instances that aren't working, that should be enough.

**Comment by Vincent-aCRM** [comment]
Attached phpinfo and IP restriction export - thank you in advance
[phpinfo.zip](https://github.com/sugarcrm/IPRestrictionManager/files/3932721/phpinfo.zip)

[IPRestrictions-notworking.xlsx](https://github.com/sugarcrm/IPRestrictionManager/files/3932733/IPRestrictions-notworking.xlsx)

**Comment by Vincent-aCRM** [comment]
Just in case it helps to compare 2 stacks, this one is another instance with different PHP version that is giving same error
[phpinfo.zip](https://github.com/sugarcrm/IPRestrictionManager/files/3932759/phpinfo.zip)

**Comment by geraldclark** [comment]
Hi Vincent,

Between the export and your screenshot, it looks like your IP is drastically different than what that validation is expecting. You have an IPv4 address in your validation but the IP in your screenshot is IPv6. To solve this, I would likely just add the IPv6 address to the list of allowed IP addresses - otherwise, you would need to investigate your network and that server to identify why it's using IPv6.

**Comment by Vincent-aCRM** [comment]
The IPv6 in the error message seems to belong to the hosting provider.
Not sure why their setup would sop his module checking IPv4 addresses
I don't suppose this module could be adapted to work with these situations, just in case IPv6 becomes more popular?

**Comment by geraldclark** [comment]
I don't believe the problem here is between IPv4 vs IPv6 validations as you should be able to set validations for either. If you are setting an IPv6 validation range and it's not working, please let me know.

The way the module works is to check for the IP addresses found in $_SERVER:
https://github.com/sugarcrm/IPRestrictionManager/blob/master/src/SugarModules/modules/IRM_IPRestrictionManager/IRM_IPRestrictionManager.php#L71

On your server, it appears that $_SERVER['REMOTE_ADDR'] is the only source for an IP which is why we are getting that address. It's trying to validate your IPv4 range against an IPv6 address. Ultimately you would need to figure out why $_SERVER['HTTP_CLIENT_IP'] isn't being set on the server. I would certainly be open to any suggestions on how else this module could obtain the requestors IP address from the server.

**Comment by Vincent-aCRM** [comment]
hi Gerald, 
I passed your last reply to the hosting company, and reply I got today was:

''Can this work on X-Forwarded-For?''

thanks again for your help on trying to solve this one, hopefully the solution will help others.

**Comment by geraldclark** [comment]
Hi Vincent,

Can you have someone at the hosting company replace the `getIpAddress()` function in 
 ./modules/IRM_IPRestrictionManager/IRM_IPRestrictionManager.php with the function in the link below and let me know if it solves the issue?

https://gist.github.com/geraldclark/c5a7765655a994f5d72d8a631dc32769

**Comment by geraldclark** [comment]
Hi Vincent,

Any updates on this?

**Comment by Vincent-aCRM** [comment]
hi @geraldclark 
sorry, didn't see you asked for an update, my apologies, all worked fine, think it was only 1 user IP v6 that was causing issue.
